modules/claims_api/lib/claims_api/v2/disability_compensation_evss_mapper.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+module ClaimsApi
+  module V2
+    class DisabilityCompensationEvssMapper
+      def initialize(auto_claim)
+        @auto_claim = auto_claim
+        @data = auto_claim&.form_data&.deep_symbolize_keys
+        @evss_claim = {}
+      end
+
+      def map_claim
+        claim_attributes
+        claim_meta
+
+        { form526: @evss_claim }
+      end
+
+      private
+
+      def claim_attributes
+        change_of_address
+        current_mailing_address
+        direct_deposit
+        disabilities
+        service_information
+        standard_claim
+        veteran_meta
+      end
+
+      def change_of_address
+        return if @data[:changeOfAddress].blank?
+
+        @evss_claim[:veteran] ||= {}
+        @evss_claim[:veteran][:changeOfAddress] = @data[:changeOfAddress]
+        @evss_claim[:veteran][:changeOfAddress].merge!({
+                                                         addressLine1: @data[:changeOfAddress][:numberAndStreet],
+                                                         addressLines2: @data[:changeOfAddress][:apartmentOrUnitNumber],
+                                                         type: 'DOMESTIC'
+                                                       })
+        @evss_claim[:veteran][:changeOfAddress][:addressChangeType] = @data.dig(:changeOfAddress, :typeOfAddressChange)
+        @evss_claim[:veteran][:changeOfAddress][:beginningDate] = @data.dig(:changeOfAddress, :dates, :beginningDate)
+        @evss_claim[:veteran][:changeOfAddress][:endingDate] = @data.dig(:changeOfAddress, :dates, :endingDate)
+        @evss_claim[:veteran][:changeOfAddress].except!(:numberAndStreet, :apartmentOrUnitNumber, :typeOfAddressChange,
+                                                        :dates)
+      end
+
+      def current_mailing_address
+        addr = @data.dig(:veteranIdentification, :mailingAddress) || {}
+        @evss_claim[:veteran] ||= {}
+        @evss_claim[:veteran][:currentMailingAddress] = addr
+        @evss_claim[:veteran][:currentMailingAddress].merge!({
+                                                               addressLine1: addr[:numberAndStreet],
+                                                               addressLines2: addr[:apartmentOrUnitNumber],
+                                                               type: 'DOMESTIC'
```

Comment:
![Screenshot 2023-06-07 at 1 13 25 PM](https://github.com/department-of-veterans-affairs/vets-api/assets/25069483/a631b07e-e4c4-484d-b64a-59413d89d806)

I feel like this could be problematic. I know this `veteran.currentMailingAddress.addressLine2` should be an apartment number, but after we map it though it just shows up as '22'.

Reply:
Should I hard code it as "Appt #{value}"? The source field `apartmentOrUnitNumber` has a max length of 5. I was originally thinking they'd put "Apt 22" for that instead, not not with a maxLength of 5.